monitoring/opentelemetry-reactive/pom.xml

```diff
@@ -27,6 +27,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-opentelemetry</artifactId>
         </dependency>
+        <dependency>
+            <!-- Required for metrics. TODO: replace when there would be another provider -->
```

Comment:
FWIW I think it is a good idea to test both OTel alone and OTel with micrometer as they are same interface betweem themselves and we need to check all combinations works. If we had this separately (added as ad hoc dependency), it would work as well and other tests here https://github.com/quarkus-qe/quarkus-test-suite/tree/main/monitoring/opentelemetry-reactive/src/test/java/io/quarkus/ts/opentelemetry/reactive would be tested without Micrometer.

Reply:
Hmm, sounds reasonable. I will try to create new test class to cover that

Reply:
Added new class